src/test/java/com/thebois/models/world/WorldTests.java

```diff
@@ -291,4 +301,151 @@ public void getTerrainsReturnsTerrainEvenIfStructuresAreOnTopOfTerrain() {
         assertThat(actualNumberOfTerrainTiles).isEqualTo(expectedNumberOfTerrainTiles);
     }
 
+    private static Stream<Arguments> getCorrectCoordinatesToTest() {
+        return Stream.of(Arguments.of(0, 0, 2, 2),
+                         Arguments.of(5, 5, 10, 10),
+                         Arguments.of(0, 0, 0, 0),
+                         Arguments.of(20, 20, 11, 11),
+                         Arguments.of(49, 49, 40, 40)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("getCorrectCoordinatesToTest")
+    public void findNearestStructureReturnsCorrect(final int startX,
+                                                   final int startY,
+                                                   final int endX,
+                                                   final int endY) {
+
+        // Arrange
+        final World world = new TestWorld(50);
+        world.createStructure(StructureType.HOUSE, endX, endY);
+
+        // Act
+        final Optional<IStructure> foundStructure = world.findNearestStructure(new Position(
+            startX,
+            startY));
+
+        // Assert
+        assertThat(foundStructure.orElseThrow().getPosition()).isEqualTo(new Position(endX, endY));
+
+    }
+
+//    private static Stream<Arguments> getInCorrectCoordinatesToTest() {
+//        return Stream.of(Arguments.of(0, 0, 60, 60),
+//                         Arguments.of(5, 5, 100, 100),
+//                         Arguments.of(0, 0, 49, 26),
+//                         Arguments.of(20, 10, 49, 49)
+//        );
+//    }
+//
+//    @ParameterizedTest
+//    @MethodSource("getInCorrectCoordinatesToTest")
+//    public void findNearestStructureReturnsIncorrect(final int startX,
+//                                                     final int startY,
+//                                                     final int endX,
+//                                                     final int endY) {
+//
+//        // Arrange
+//        final World world = new TestWorld(50);
+//        world.createStructure(StructureType.HOUSE, endX, endY);
+//
+//        // Act
+//        final Optional<IStructure> foundStructure = world.findNearestStructure(new Position(
+//            startX,
+//            startY));
+//
+//        // Assert
+//        assertThat(foundStructure.isEmpty()).isTrue();
+//    }
+
+    @Test
+    public void returnsNoNearestStructure() {
```

Comment:
style

When does it return no nearest structure?

`returnsNoNearestStructureWhenWorldIsEmpty` is more descriptive.